src/workerd/api/web-socket.c++

```diff
@@ -471,12 +507,23 @@ void WebSocket::startReadLoop(jsg::Lock& js) {
                 (jsg::Lock& js, kj::Maybe<kj::Exception>&& maybeError) mutable {
     auto& native = *farNative;
     KJ_IF_MAYBE(e, maybeError) {
-      if (!native.closedIncoming && e->getType() == kj::Exception::Type::DISCONNECTED) {
+      WebSocketProtocolError wspe;
+      if (wspe.decodeFromException(*e)) {
+        // The client sent us an invalid websocket message.
+        if (!native.closedOutgoing) {
+          // Send a close message to the client if we can.
+          close(js, wspe.getCode(), kj::str(wspe.getDescription()));
+        }
+        // Report to the application as an error event.
+        dispatchEventImpl(js, jsg::alloc<ErrorEvent>(
+          kj::str("WebSocket code ", wspe.getCode()),
+          jsg::Value(js.v8Isolate, js.wrapString(wspe.getDescription())), js.v8Isolate));
```

Comment:
```suggestion
          js.v8Ref(js.wrapString(wspe.getDescription())), js.v8Isolate);
```

Reply:
This makes the ErrorEvent constructor unhappy: https://gist.github.com/smerritt/6a71579554997f7e7295078bb2817a8a

I don't have much experience with workerd code, so I'm not sure what the best way to wrap a kj::String as a jsg::Value is.